src/goto-instrument/nondet_volatile.cpp

```diff
@@ -246,9 +279,9 @@ void nondet_volatilet::typecheck_model(
 }
 
 void nondet_volatilet::typecheck_options(
-  const goto_modelt &goto_model,
   const optionst &options)
 {
+  PRECONDITION(!all_nondet);
```

Comment:
:pick:  There's a lot happening in 782395c0457cd970c07266a337f5d0dde3b3281d which makes it a bit hard to recognise as a refactoring (i.e. there's so much code moving around its a bit hard to see where what is moving).